File: 1578/solution.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        i = 0
        res = 0
        n = len(colors)
        while i<n:
            total = neededTime[i]
            maxi = total
            while i+1<n and colors[i]==colors[i+1]:
                i+=1
                total+=neededTime[i]
                maxi = max(maxi,neededTime[i])
            i+=1
            res+=(total-maxi)
        return res

File: 1578/solution.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int i = 0, result=0, total, max;
        int n = colors.length();
        while(i<n){
            total = neededTime[i];
            max = neededTime[i];
            while(i+1<n && colors.charAt(i)==colors.charAt(i+1)){
                total+=neededTime[++i];
                max = Math.max(max,neededTime[i]);
            }
            result+=(total-max);
            i++;
        }
        return result;        
    }
}
